### DISCUSSION TITLE
"Question About SHR Operation in Lesson 14 - Patrick's Explanation"

### DISCUSSION THREAD

**Discussion by Basha-dude** [Cyfrin/assembly-evm-opcodes-and-formal-verification-course#14] [General]
@PatrickAlphaC, in [Lesson 14 - SHR](https://updraft.cyfrin.io/courses/formal-verification/horse-store/shr?lesson_format=video), you mentioned that for the value `0x0102` in binary (`0b100000010`), performing a `shr 2`  operation results in `0b01000000.` However, in the written lesson, it is correctly stated as `0b001000000`. It seems there was a mistake in the video explanation. Could you please confirm and correct this?

**Comment by EngrPips** [comment]
Thanks for pointing out that friend. I trust this will be attended to by him as soon as he is available.

**Reply by Basha-dude** [reply]
Thanks for recognizing it.

**Reply by EngrPips** [reply]
You are always welcome, Friend.

**Comment by Basha-dude** [comment]
HEY @EngrPips , when he is doing hex and dec.  **it is giving correct answer**

**Reply by EngrPips** [reply]
Hello @Basha-dude, I am trying to understand what you mean.

**Reply by Basha-dude** [reply]
Hey @EngrPips,

When @PatrickAlphaC performed a **shr 2** operation on the value `0x0102`, the binary representation `0b100000010` shifted right by 2 results in `0b01000000`. However, in the written lesson, it is shown as `0b001000000`.

Despite the difference in binary representation, both `0b01000000` and `0b001000000` represent the same value. **Converting these to hexadecimal and decimal yields the same results (0x40 and 64, respectively)**.

**Comment by EngrPips** [comment]
Hey @Basha-dude, Now I understand your point, and yeah, they yield the same result because we padded with 0 from the left, so one `0` and hundred `0` is still `0,` but then it is better we do a proper shifting in situations where we have values that are not zero after we finish our shifting. For example 

0b111100000

Shifting the above value two times to the right would result in 0b001111000, but moving by one bit to the right would result in 0b011110000. You would see that those two values are not the same.

**Reply by Basha-dude** [reply]
hey @EngrPips, do some explain that it did not understand the **first question** i started in this discussion

**Reply by EngrPips** [reply]
Hello @Basha-dude, I am not sure what you mean.

**Reply by Basha-dude** [reply]
hello  @EngrPips , i don't understand whole thing `shr 2` , how it shifted to right ,please explain it with a simple example

**Reply by Basha-dude** [reply]
hey @EngrPips, thanks for helping it was **very helpful**

**Reply by EngrPips** [reply]
I am glad that helped.

**Comment by EngrPips** [comment]
Oh, I see, You should watch this `https://www.youtube.com/watch?v=T7Bw5WqvhHY` YouTube video it explains bit shifting as a whole.

**Reply by EngrPips** [reply]
> Oh, I see. You should watch this `https://www.youtube.com/watch?v=T7Bw5WqvhHY` YouTube video. It explains bit shifting as a whole.

If after you watch the video, something isn't clear, please come back here and let us help clarify things more.
